### PR TITLE
Document additional options for String.split/3

### PR DESCRIPTION
**Pull request by jbranchaud**
Issue #5213

### CODE REVIEW

**Comment by josevalim** [comment]
All yours @whatyouhide!

File: lib/elixir/lib/string.ex [hunk 1]
```diff
@@ -276,6 +276,10 @@ defmodule String do
   Empty strings are only removed from the result if the
   `trim` option is set to `true` (default is `false`).
 
+  When the pattern used is a regular expression, the string will be
+  split using `Regex.split/3`. This means that it will accept
```
Comment (whatyouhide) [inline lib/elixir/lib/string.ex]: Let's say "This means that <strike>it</strike>this function <strike>will accept</strike>acceots [...]", wdyt? We usually use present verbs and I feel "it" is not superclear here. :)

Comment (jbranchaud) [inline lib/elixir/lib/string.ex]: Cool, makes sense. Thanks for the feedback.

**Comment by jbranchaud** [comment]
@whatyouhide I incorporated your suggestions, let me know what you think :grin:

**Comment by whatyouhide** [comment]
Thanks a lot @jbranchaud! 💟

**Pull request merged**